pkg/command/plugin.go

```diff
@@ -203,16 +211,46 @@ func newInstallPluginCmd() *cobra.Command {
 	var installCmd = &cobra.Command{
 		Use:   "install [name]",
 		Short: "Install a plugin",
-		Args:  cobra.ExactArgs(1),
+		Args:  cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			var err error
-
-			pluginName := args[0]
+			var pluginName string
 
 			if !configtypes.IsValidTarget(targetStr, true, true) {
 				return errors.New("invalid target specified. Please specify correct value of `--target` or `-t` flag from 'kubernetes/k8s/mission-control/tmc'")
 			}
 
+			if !config.IsFeatureActivated(constants.FeatureCentralRepository) {
+				return legacyPluginInstall(cmd, args)
```

Comment:
I had to split the old and new behaviours into two functions because the linter reported it had gotten too complex.